irohad/torii/processor/impl/query_processor_impl.cpp

```diff
@@ -72,6 +90,32 @@ namespace iroha {
           std::make_shared<shared_model::proto::QueryResponse>(
               qry_resp->getTransport()));
     }
+    rxcpp::observable<
+        std::shared_ptr<shared_model::interface::BlockQueryResponse>>
+    QueryProcessorImpl::blocksQueryHandle(
+        std::shared_ptr<shared_model::interface::BlocksQuery> qry) {
+      if (not checkSignatories(*qry)) {
+        auto response = buildBlocksQueryError("wrong signatories");
+        return rxcpp::observable<>::just(
+            std::shared_ptr<shared_model::interface::BlockQueryResponse>(
+                response));
+      }
+      const auto &wsv_query = storage_->getWsvQuery();
+      auto qpf = QueryProcessingFactory(wsv_query, storage_->getBlockQuery());
+      if (not qpf.validate(*qry)) {
+        auto response = buildBlocksQueryError("stateful invalid");
+        blocksQuerySubject_.get_subscriber().on_next(response);
+        return rxcpp::observable<>::just(
+            std::shared_ptr<shared_model::interface::BlockQueryResponse>(
+                response));
+      }
+      storage_->on_commit().subscribe(
```

Comment:
I thinks this should happen in constructor, otherwise we might send same blocks every time we subscribe to the get blocks

Reply:
yes, me stupid